### PR TITLE
Use Elem to extract finders out of classes

### PR DESCRIPTION
**Pull request by connorshea**
I asked this in Slack:

>is there a reason we can't just move some of the methods on models' relation classes up to ActiveRecord::Base?
>
>```ruby
># broke
>class Potion::ActiveRecord_Relation < ActiveRecord::Relation
>  Elem = type_member(fixed: Potion)
>
>  sig { params(args: T.untyped).returns(Potion) }
>  def find(*args); end
>
>  sig { params(args: T.untyped).returns(T.nilable(Potion)) }
>  def find_by(*args); end
>
>  sig { params(args: T.untyped).returns(Potion) }
>  def find_by!(*args); end
>end
>
># woke
>class ActiveRecord::Relation
>  sig { params(args: T.untyped).returns(Elem) }
>  def find(*args); end
>
>  sig { params(args: T.untyped).returns(T.nilable(Elem)) }
>  def find_by(*args); end
>
>  sig { params(args: T.untyped).returns(Elem) }
>  def find_by!(*args); end
>end
>```

So I tested it, and it turns out that Elem does in fact seem to work correctly 🎉 

We could probably move `exists?`, `any?`, `many?`, `none?`, and `one?` up as well. And probably also `map`?

### CODE REVIEW

**Comment by connorshea** [comment]
I tried adding `any?`, `many?`, `none?`, and `one?`, but I'm getting some typechecking errors due to method redefinition :/

```
sorbet/rbi/gems/activerecord.rbi:2527: Method `ActiveRecord::Relation#any?` redefined without matching argument count. Expected: `1`, got: `0` https://srb.help/4010
    2527 |  def any?; end
            ^^^^^^^^
    sorbet/rails-rbi/active_record_relation.rbi:69: Previous definition
    69 |  def any?(*args); end
          ^^^^^^^^^^^^^^^

sorbet/rbi/gems/activerecord.rbi:2562: Method `ActiveRecord::Relation#many?` redefined without matching argument count. Expected: `1`, got: `0` https://srb.help/4010
    2562 |  def many?; end
            ^^^^^^^^^
    sorbet/rails-rbi/active_record_relation.rbi:72: Previous definition
    72 |  def many?(*args); end
          ^^^^^^^^^^^^^^^^

sorbet/rbi/gems/activerecord.rbi:2565: Method `ActiveRecord::Relation#none?` redefined without matching argument count. Expected: `1`, got: `0` https://srb.help/4010
    2565 |  def none?; end
            ^^^^^^^^^
    sorbet/rails-rbi/active_record_relation.rbi:75: Previous definition
    75 |  def none?(*args); end
          ^^^^^^^^^^^^^^^^

sorbet/rbi/gems/activerecord.rbi:2566: Method `ActiveRecord::Relation#one?` redefined without matching argument count. Expected: `1`, got: `0` https://srb.help/4010
    2566 |  def one?; end
            ^^^^^^^^
    sorbet/rails-rbi/active_record_relation.rbi:78: Previous definition
    78 |  def one?(*args); end
          ^^^^^^^^^^^^^^^
Errors: 4
```

**Comment by connorshea** [comment]
oddly, when I make this change to the `ActiveRecord::AssociationRelation` classes it causes typecheck errors 🤔

File: lib/bundled_rbi/active_record_relation.rbi [hunk 1]
```diff
@@ -0,0 +1,79 @@
+# typed: strong
+class ActiveRecord::Relation
+  sig { params(args: T.untyped).returns(Elem) }
```
Comment (manhhung741) [inline lib/bundled_rbi/active_record_relation.rbi]: Interesting that it can find that `Elem` is defined.

**Comment by connorshea** [comment]
FWIW I ran sorbet-rails' generator with this branch, as well as `srb init`, and it worked fine for [vglist](https://github.com/connorshea/VideoGameList). I'll add tests.

**Comment by manhhung741** [comment]
Thanks, @connorshea! I'm ready to release a new version of Sorbet-Rails after this PR & my PR is merged!

**Comment by connorshea** [comment]
@manhhung741 done :)

**Comment by manhhung741** [comment]
Thank you :-)

**Pull request merged**